src/EmojiList.js

```diff
@@ -1,19 +1,28 @@
 import React, { Component } from 'react';
 import { connect } from 'react-redux';
 import ItemComponent from './EmojiComponent';
+import SettingsModeButtons from './SettingsModeButtons';
 
 const apply = connect(
-    state => ({ smiles: state.items.smiles })
+    state => ({
+        smiles: state.items.smiles,
+        mode: state.pages.mode
+    })
 )
 
 class EmojiList extends Component{
 
     render() {
-        console.log("EmojiList ", this.props);
+        let { smiles, mode } = this.props;
         return(
             <div>
-                {this.props.smiles.map(item => {
-                    return <ItemComponent text={item.text} id={item.id} />
+                {smiles.map(item => {
+                    return (
+                        <div class='buttons-line' key={item.id}>
+                            <ItemComponent text={item.text}  />
```

Comment:
It is possible solution, but I would prefer to make it like this:
```javascript
smiles.map(item => mode ? (
  <ItemComponent
    key={item.id}
    text={item.text}
  />
) : (
  <div
    className='buttons-line' // don't forget about class -> className transformation
    key={item.id}
  >
    <ItemComponent
      text={item.text}
    />
    <SettingsModeButtons
      id={item.id}
    />
  </div>
));
```
Or you can even move `ItemComponent` to separate variable in `map` function and use it without additional code:
```javascript
smiles.map(item => {
  const item = (
    <ItemComponent
      key={item.id}
      text={item.text}
    />
  );

  return mode ? item : (
    <div
      className='buttons-line'
      key={item.id}
    >
      {item}
      <SettingsModeButtons
        id={item.id}
      />
    </div>
  )
});
```

Reply:
I left my own solution